js/interface.js

```diff
@@ -234,6 +237,23 @@ $.each(externalAppValueMap, function(key) {
   });
 });
 
+Fliplet.Studio.onMessage(function(event) {
+  if (event.data) {
+    switch (event.data.event) {
+      case 'page-field-error':
```

Comment:
If `options.required` is provided in the parameter for the link provider, can the link provider check that a screen is selected on submission, instead of waiting for `page-field-error` to be sent from the provider caller?

I feel that the link provider should be able to handle the validation within itself.

Reply:
@tonytlwu There is an issue with that approach.
When we validate the page field from the link provider we encounter several issues.

1. We still need to fire an event from the provider to highlight accordion and text.
2. In order to save data correctly we still need to monitor the page value from the provider
3. When chat is set correctly save function is fired and the provider is closing - [screencast](https://share.getcloudapp.com/NQuovLPL)

The first two issues aren't critical in my opinion but the third one is.
And at the time I can't figure out how to solve it except separate validation and save.
What I mean, that when we do validation in the LFD we should emit the event into the link provider.
And when validation is passed we call forwardSaveRequest.

In light of this situation could we stay with the current solution or there is another way we could work around it?

Reply:
@YaroslavOvdii Yes we can go with the current solution for now.